src/Tribe/Views/V2/Template/Excerpt.php

```diff
@@ -63,37 +62,27 @@ public function maybe_filter_excerpt_length( $length ) {
 	 * @return string The excerpt read more link modified, if necessary.
 	 */
 	public function maybe_filter_excerpt_more( $link ) {
-		/**
-		 * Because `maybe_filter_excerpt_more` needs to call `tribe_get_event` it creates
-		 * a infinite loop with adding excerpt calls again.
-		 */
-		remove_filter( 'excerpt_more', [ tribe( Hooks::class ), 'filter_excerpt_more' ], 50 );
-
 		if ( is_admin() ) {
 			return $link;
 		}
 
-		$event = tribe_get_event( get_the_ID() );
-
-		/**
-		 * Because `maybe_filter_excerpt_more` needs to call `tribe_get_event` it creates
-		 * a infinite loop with adding excerpt calls again.
-		 */
-		add_filter( 'excerpt_more', [ tribe( Hooks::class ), 'filter_excerpt_more' ], 50 );
+		// This method could be called in the context of a call to `tribe_get_event`; here we only need the permalink.
```

Comment:
My issue with this is if someone needs other information when using this particular template and uses `tribe_get_event()` inside of that component when trying to extend it will throw an infinite loop. I would much rather provide the full object even if it's a small performance hit.

I don't know how big of a performance hit this is tho.

Reply:
It's a huge performance hit, not a small one: it recursively invalidates the caching of the `tribe_get_event` function.
I can take another look at this, and I agree with you on how we might need the "full" object in future iterations.

I will update this PR if the work required to get around this is not a mountain.

Reply:
I've pushed a [last commit](https://github.com/moderntribe/the-events-calendar/pull/2889/commits/32a88c6a647b223dee5459cc25b260856422a27c) to this PR to hide the excerpt in a lazy string. This solves both issues.